Allure.Net.Commons/Functions/FormatFunctions.cs

```diff
@@ -37,6 +43,23 @@ public static string Format(object? value)
             return formatter.Format(value);
         }
 
-        return JsonConvert.SerializeObject(value);
+        try
+        {
+            return JsonConvert.SerializeObject(
+                value,
+                Formatting.Indented,
+                new JsonSerializerSettings
+                {
+                    ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                    Error = (_, args) =>
+                    {
+                        args.ErrorContext.Handled = true;
+                    }
+                });
+        }
+        catch
+        {
+            return JsonConvert.Undefined;
```

Comment:
Let's fall back to `ToString` here. At least we'll get different values for different types (or even something useful if this method is overridden).
```suggestion
            return value.ToString();
```